### PR TITLE
issue #763 LCO List Component based on new Figma design

### PR DESCRIPTION
**Pull request by actuallyyun**
#### Empty State 
- web
Screenshot 2023-04-19 at 11 47 35

- mobile
Screenshot 2023-04-19 at 11 47 52

#### Happy Path
- web
Screenshot 2023-04-19 at 11 53 49

- mobile
Screenshot 2023-04-19 at 11 54 08

-Tooltip

Screenshot 2023-04-24 at 10 46 59

### CODE REVIEW

**Comment by actuallyyun** [comment]
I had a close look at `MobileDialog` component, `DialogTrigger` element is a button, which does not work well with the new Figma design. Here the trigger is the whole LCO banner. Is it possible to customize `DialogTrigger` and make it behave like a container, or should I implement `Dialog` separately without using the existing components? What are your thoughts on this? @vnugent

**Comment by vnugent** [comment]
> I had a close look at `MobileDialog` component, `DialogTrigger` element is a button, which does not work well with the new Figma design. Here the trigger is the whole LCO banner. Is it possible to customize `DialogTrigger` and make it behave like a container, or should I implement `Dialog` separately without using the existing components? What are your thoughts on this? @vnugent

Let me have look at the Figma and then I can give you a better feedback.  In general we're deprecating the `headlessui` library.  Therefore please avoid using components that are based on that library.

>  Is it possible to customize `DialogTrigger` and make it behave like a container

If you use Radix UI trigger then Yes you can customize the trigger using your own component.  Look into the `asChild` property.

Ps: also when ready please share some screen shots

**Comment by actuallyyun** [comment]
> `headlessui` library. Therefore please avoid using components that are based on that library.

Super good to know! Should I try to stay with `Radix UI` then?
Will add screenshot, thanks for the reminder.

**Comment by vnugent** [comment]
Regarding implementation for trigger it depends on whether we decide the LCO profile card to be a hover card (open on hover) or a modal popup (open on click). 

Add area modal

Screenshot 2023-04-18 at 4 32 42 PM

https://github.com/OpenBeta/open-tacos/blob/develop/src/components/edit/Triggers.tsx#L95

---

I think for the scope of this PR, you can simply let users click on the LCO website link and we can figure out the LCO card in another PR.

LCO icon placeholder: the [users](https://heroicons.com/) hero icon will do for now
```javascript
import {  UsersIcon } from '@heroicons/react/24/outline'
```

**Comment by actuallyyun** [comment]
Cool I will finish up the UI. How about the info icon tooltip? Should I leave it out for now as well? @vnugent

**Comment by vnugent** [comment]
Can you fix linting errors?  I can merge it and you can do a follow up PR for tool tips and other things.

**Comment by actuallyyun** [comment]
Ok getting on it now

**Comment by actuallyyun** [comment]
> Can you fix linting errors? I can merge it and you can do a follow up PR for tool tips and other things.

fixed!

**Comment by actuallyyun** [comment]
Should I undo the lco test data before you merge it?

**Comment by vnugent** [comment]
> Should I undo the lco test data before you merge it?

yes please.  We want the existing LCO info to continue to work.

**Comment by actuallyyun** [comment]
Removed tooltip and test data.

File: src/components/lco/PageBanner.tsx [hunk 1]
```diff
@@ -36,62 +44,93 @@ type PageBannerProps = Pick<ClimbType, 'ancestors'>
...
-  if (orgs.length === 0) return null
   return (
-    <div className='mt-8 bg-area-cue/20 p-6 rounded-box'>
-      <div className='mt-2 mb-1 text-xs text-base-300'>Local climbing organizations</div>
-      <div className='flex flex-col'>
-        {orgs.map(orgProfile => <IndividualBanner key={orgProfile.id} profile={orgProfile} />)}
+    <div className='grid pt-6 pb-4 lg:pb-16 lg:pt-16'>
+      <div className='col-span-full flex justify-start items-center pb-6'>
+        <h3 className='mr-4'>Local climbing organizations</h3>
+        <InformationCircleIcon className='h-6 w-6' />
```
Comment (vnugent) [inline src/components/lco/PageBanner.tsx]: Why don't we just add a tool tip and link to
"Learn more about our [initiative](https://openbeta.substack.com/p/openbeta-and-lcos)."

### Tooltip usage: 

- Tooltip with a simple string content: https://github.com/OpenBeta/open-tacos/blob/7db52ce0c7883c79cb4ee9b365c9de784da4aa88/src/components/edit/form/AreaDesignationRadioGroup.tsx#L36
- Tooltip with JSX content: https://github.com/OpenBeta/open-tacos/blob/develop/src/components/edit/form/LegacyFAInput.tsx

File: src/components/lco/PageBanner.tsx [hunk 2]
```diff
@@ -36,62 +44,93 @@ type PageBannerProps = Pick<ClimbType, 'ancestors'>
...
-  </HoverCard.Root>
+  <>
+    <div className='sm:inline-block mr-6 mb-6'>
+      <div className='flex items-center bg-light hover:bg-on-hover pl-5 pr-7 rounded-2xl'>
+        <UsersIcon className='h-10 w-10' />
+        <div className='py-5  pl-4 overflow-hidden'>
+          <p className='text-base leading-6'>{profile.name}</p>
+
+          <p className='text-xs underline'>
+            <a href={profile.website}>{profile.website}</a>
```
Comment (vnugent) [inline src/components/lco/PageBanner.tsx]: can you add `_target="blank"` to open LCO website in a new tab?

File: src/components/lco/PageBanner.tsx [hunk 3]
```diff
@@ -36,62 +44,93 @@ type PageBannerProps = Pick<ClimbType, 'ancestors'>
...
-    </HoverCard.Trigger>
-    <HoverCard.Portal>
-      <HoverCard.Content sideOffset={10} side='top' className='z-40'>
-        <Card profile={profile} />
-      </HoverCard.Content>
-    </HoverCard.Portal>
-  </HoverCard.Root>
+  <>
+    <div className='sm:inline-block mr-6 mb-6'>
+      <div className='flex items-center bg-light hover:bg-on-hover pl-5 pr-7 rounded-2xl'>
```
Comment (vnugent) [inline src/components/lco/PageBanner.tsx:82]: Highlighting (darkening) of the LCO item on hover is a bit confusing to me as I gives me the impression that I can click on the whole div, but only the url is clickable.

**Review by vnugent — CHANGES_REQUESTED** [review]

File: src/components/lco/PageBanner.tsx (hunk 3 above)
Comment (actuallyyun) [reply src/components/lco/PageBanner.tsx:82]: Yeah I think AJ's intention is to let the user click on the whole div and show the LCO profile card.

File: src/components/lco/PageBanner.tsx (hunk 2 above)
Comment (actuallyyun) [reply src/components/lco/PageBanner.tsx]: Sure!

File: src/components/lco/PageBanner.tsx [hunk 4]
```diff
@@ -36,62 +44,101 @@ type PageBannerProps = Pick<ClimbType, 'ancestors'>
...
-      <ContentTrigger profile={profile} />
-    </HoverCard.Trigger>
-    <HoverCard.Portal>
-      <HoverCard.Content sideOffset={10} side='top' className='z-40'>
-        <Card profile={profile} />
-      </HoverCard.Content>
-    </HoverCard.Portal>
-  </HoverCard.Root>
+  <>
+    <div className='sm:inline-block mr-6 mb-6'>
```
Comment (vnugent) [inline src/components/lco/PageBanner.tsx:81]: what is the intention here? `sm:inline-block`

I understand this as for  anything smaller `sm` don't make it an inline-block (default = block).  Anything sm, md, lg and larger -\> inline-block

https://tailwindcss.com/docs/responsive-design#working-mobile-first

Comment (actuallyyun) [reply src/components/lco/PageBanner.tsx:81]: Good catch.
In my experiment,  for the web layout to work, each div needs to be `inline-block` when using grid layout( in my understanding at least). On mobile, a regular grid layout will do(each div takes a row). So my intention was to only change the display to `inline-block` on web(`sm` breakpoints and onwards). I wasn't targeting mobile with `sm:inline-block`

**Comment by vnugent** [comment]
@actuallyyun let me know when it's ready to merge.  You'll need to resolve some conflicts in cragSummary.tsx

**Comment by actuallyyun** [comment]
> 1

@vnugent ready to merge!

**Comment by vnugent** [comment]
Did the build and linter pass locally in your env?  (you can commit without --no-verify flag)

https://github.com/OpenBeta/open-tacos/actions/runs/4820193159/jobs/8584351722?pr=782#step:5:10

**Comment by actuallyyun** [comment]
Sorry I must have reversed the linting errors I had fixed earlier when I resolved the merge conflicts. It should be fixed now.

**Comment by vnugent** [comment]
@zichongkao  I think recent changes to the LCO history object breaks the frontend build

https://github.com/OpenBeta/open-tacos/actions/runs/4829298519/jobs/8604171055?pr=782#step:8:43

**Comment by zichongkao** [comment]
Thanks for pointing this out! I'll take a look and put in a fix by today!

**Comment by zichongkao** [comment]
This should fix it: https://github.com/OpenBeta/open-tacos/pull/805 together with https://github.com/OpenBeta/openbeta-graphql/pull/276. But we need the openbeta-graphql changes to be deployed so the open-taco change can be built before we merge it. Then I'll rebase this PR onto these changes!

**Comment by vnugent** [comment]
#805 merged.  @zichongkao or @actuallyyun can you please rebase this PR then we're good to merge. Thank you!

**Comment by zichongkao** [comment]
@actuallyyun This is actually a branch on your repo, so you'll have to rebase -- I can't do it. Sorry for the trouble!

**Comment by actuallyyun** [comment]
Rebased and pushed

**Pull request merged**